### PR TITLE
Variants section added to button page

### PR DESCRIPTION
**Pull request by halocline**
Related Issues: https://github.com/orgs/hpe-design/projects/1#card-35491931

[Preview](https://deploy-preview-629--keen-mayer-a86c8b.netlify.com/components/button)

Work Included:
- Added Variants section to `button.js`, plus added button variant examples
- Simplified `<ContentSection>` as new designs have reduced the nuance. Recommend keeping as component for spacing and scanability reasons.
- Minor tweaks to `<Example>` to adjust style on smaller screens
- Allow for subsection heading text to be semantically correct but render at a different size

For Discussion:
- Variants seemed buried below Guidance and may get lost
- Button examples do not demonstrate variants such as:
  - disabled
  - icon
  - sizes
- Sizing of the `<Example>` container seems quite large and introduces a lot of scrolling, especially if there are additional variants we'd like to show (e.g. disabled, icon, sizing). I am going to create another branch with smaller height <Example> so we can contrast with this branch.

### CODE REVIEW

File: aries-site/src/examples/components/button/ButtonExample.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import { Button } from 'grommet';
+import { Next } from 'grommet-icons';
+
+export const ButtonExample = () => {
+  return <Button label="Button" icon={<Next />} reverse onClick={() => {}} />;
```
Comment (taysea) [inline aries-site/src/examples/components/button/ButtonExample.js:6]: It doesn't feel "default" if we have to use `reverse` on it. I know this is adhering to designs but I'm wondering if it's not the best example for default.

**Comment by halocline** [comment]
agree. I flip-flopped on it. Will take it out.

File: aries-site/src/layouts/content/ContentSection.js [hunk 2]
```diff
@@ -1,37 +1,16 @@
...
+  <Box
+    as="section"
+    align="start"
+    gap="medium"
+    pad={{
+      bottom: 'medium',
+      horizontal: 'xxsmall',
+      top: 'large',
+    }}
+    {...rest}
```
Comment (taysea) [inline aries-site/src/layouts/content/ContentSection.js:14]: Love seeing the cleanup in this.

File: aries-site/src/layouts/content/Example.js [hunk 3]
```diff
@@ -93,6 +94,7 @@ export const Example = ({
   ...rest
 }) => {
   const theme = React.useContext(ThemeContext);
+  const viewport = React.useContext(ResponsiveContext);
```
Comment (taysea) [inline aries-site/src/layouts/content/Example.js:97]: Is there a reason we are using `viewport` as opposed to `size` like we usually have been doing?

**Comment by halocline** [comment]
size was already declared as a variable elsewhere

File: aries-site/src/layouts/content/Example.js (hunk 3 above)
Comment (halocline) [reply aries-site/src/layouts/content/Example.js:97]: size was already declared as a variable elsewhere, line 104 I believe

File: aries-site/src/examples/components/button/ButtonExample.js (hunk 1 above)
Comment (halocline) [reply aries-site/src/examples/components/button/ButtonExample.js:6]: agree. I flip-flopped on it. Will take it out.

File: aries-site/src/layouts/content/Example.js (hunk 3 above)
Comment (taysea) [reply aries-site/src/layouts/content/Example.js:97]: What about destructing that line to be: 
```
const {
    size: { medium, small },
  } = defaultProps.theme.global;
```
Then, we can reserve `size` for ResponsiveContext you're adding

Comment (taysea) [reply aries-site/src/layouts/content/Example.js:97]: or maybe better:
```
  const { medium, small } = defaultProps.theme.global.size;
```

Comment (halocline) [reply aries-site/src/layouts/content/Example.js:97]: 👍

File: aries-site/src/layouts/content/ExampleControls.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ const ControlButton = ({ children, ...rest }) => (
     {({ hover }) => (
       <Box
         direction="row"
+        align="center"
```
Comment (taysea) [inline aries-site/src/layouts/content/ExampleControls.js:11]: On mobile, I think I'd expect the "see full screen" button to be at the top of the right. What do you think?
Screen Shot 2020-04-01 at 3 45 34 PM

**Pull request merged**